src/Parser.zig

```diff
@@ -1394,12 +1394,24 @@ fn gnuAttribute(p: *Parser) !bool {
     return true;
 }
 
-/// attributeSpecifier : (keyword_attribute '( '(' attributeList ')' ')')*
 fn attributeSpecifier(p: *Parser) Error!void {
+    return attributeSpecifierExtra(p, null);
+}
+
+/// attributeSpecifier : (keyword_attribute '( '(' attributeList ')' ')')*
+fn attributeSpecifierExtra(p: *Parser, declarator_name: ?TokenIndex) Error!void {
     while (true) {
         if (try p.gnuAttribute()) continue;
         if (try p.c2xAttribute()) continue;
-        if (try p.msvcAttribute()) continue;
+        const maybe_declspec_tok = p.tok_i;
+        if (try p.msvcAttribute()) {
+            if (declarator_name) |name_tok| {
+                try p.errTok(.declspec_not_allowed_after_declarator, maybe_declspec_tok);
+                try p.errTok(.declarator_name_tok, name_tok);
+                p.attr_buf.len = p.attr_buf.len - 1;
```

Comment:
I don't think this is correct, `msvcAttribute` can add multiple attributes so it should save the len before calling it.